File: backend/api/src/main/scala/com/github/ai/split/api/request/PostMemberRequest.scala
package com.github.ai.split.api.request

import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder}

case class PostMemberRequest(
  groupUid: String,
  name: String
)

object PostMemberRequest {
  implicit val encoder: JsonEncoder[PostMemberRequest] = DeriveJsonEncoder.gen[PostMemberRequest]
  implicit val decoder: JsonDecoder[PostMemberRequest] = DeriveJsonDecoder.gen[PostMemberRequest]
}

File: backend/api/src/main/java/com/github/ai/split/api/request/PostMemberRequest.java
package com.github.ai.split.api.request;

public class PostMemberRequest {

    public String groupUid = "";
    public String name = "";

    public PostMemberRequest() {
    }

    public PostMemberRequest(String groupUid, String name) {
        this.groupUid = groupUid;
        this.name = name;
    }
}
